src/plugins/knex.js

```diff
@@ -49,41 +55,37 @@ function createWrapRunnerQuery (tracer, config, usePromise) {
       setDBTags(this, span)
 
       return scope.activate(span, () => {
-        if (usePromise) {
-          return new Promise((resolve, reject) => {
-            // we can't use then.catch.finally because finally is not supported
-            // on node 8 which knex still supports.
-            const that = this
-            original.apply(this, arguments)
-              .then(function () {
-                resolve.apply(that, arguments)
-                span.finish()
-              })
-              .catch(function (e) {
-                addError(span, e)
-                reject.apply(that, arguments)
-                span.finish()
-              })
-          })
-        } else {
-          try {
-            return original.apply(this, arguments)
-          } catch (e) {
-            throw addError(span, e)
-          } finally {
-            span.finish()
```

Comment:
We added promise based handling for newer version of knex some time ago and kept try..catch based handling around for older versions. Turns out promise based handling actually works with all versions.